pkg/controller/nodetune/tune.go

```diff
@@ -110,37 +71,35 @@ func scyllaDataDirMountHostPaths(ctx context.Context, criClient cri.Client, scyl
 	return dataDirs.List(), nil
 }
 
-func cpusetFromCRI(ctx context.Context, client cri.Client, cid string) (cpuset.CPUSet, error) {
-	cs, err := client.Inspect(ctx, cid)
+func cpusetFromKubelet(ctx context.Context, podResourcesClient kubelet.PodResourcesClient, scyllaPod *corev1.Pod, containerName string) (cpuset.CPUSet, error) {
+	prs, err := podResourcesClient.List(ctx)
 	if err != nil {
-		return cpuset.CPUSet{}, fmt.Errorf("can't inspect container %q, %w", cid, err)
+		return cpuset.CPUSet{}, fmt.Errorf("can't list pod resources: %w", err)
 	}
 
-	if cs.Info.RuntimeSpec == nil {
-		klog.V(2).InfoS("No container status available", "ContainerID", cid)
-		return cpuset.CPUSet{}, nil
-	}
+	for _, pr := range prs {
+		if pr.Namespace != scyllaPod.Namespace || pr.Name != scyllaPod.Name {
+			continue
+		}
 
-	containerCpuSet, err := cpuset.Parse(cs.Info.RuntimeSpec.Linux.Resources.CPU.Cpus)
-	if err != nil {
-		return cpuset.CPUSet{}, fmt.Errorf("can't parse container %q cpuset %q, %w", cid, cs.Info.RuntimeSpec.Linux.Resources.CPU.Cpus, err)
-	}
+		cr, _, ok := slices.Find(pr.Containers, func(cr *v1.ContainerResources) bool {
+			return cr.Name == containerName
+		})
+		if !ok {
+			continue
+		}
 
-	return containerCpuSet, nil
-}
+		asInts := slices.ConvertSlice(cr.CpuIds, func(cpuid int64) int {
```

Comment:
nit: `asInts` is kind of missing "what as ints" part, `cpuIDs` or `cpuIDInts` might fit better